gslib/utils/text_util.py

```diff
@@ -148,10 +149,8 @@ def DecodeLongAsString(long_to_convert):
   Returns:
     String decoded from the input long.
   """
-  if isinstance(long_to_convert, basestring):
-    # Already converted.
-    return long_to_convert
-  return hex(long_to_convert)[2:-1].decode('hex')
+  unhexed = binascii.unhexlify(hex(long_to_convert)[2:].rstrip('L'))
```

Comment:
Don't forget to import binascii :)

Reply:
Whoops! Added that in. 👍 

What's a good way to test these code paths btw? Integ tests with XML api preferred and AWS credentials still seemed to miss my bugs, and I was able to successfully cp and cat s3 buckets when I cut the PR. Maybe my boto is misconfigured though if it's missing these?

Reply:
You'll need to run `test` with the `-s` argument to test against S3, e.g. `BOTO_CONFIG=/path/to/boto.cfg ./gsutil test -s -p 30`

Reply:
Oh, I didn't know about the `-s` flag for test! Thanks! 😄